### PR TITLE
JVM Parameter Changes

### PR DESCRIPTION
**Pull request by lswank**
As per the discussion in the weekly meeting, I am creating this draft pull request to enable per-line commenting for Ben, Robert, and Linkous. Be sure to add a ?w=1 to the end of the URL to ignore whitespace changes.

### CODE REVIEW

File: adjust [hunk 1]
```diff
@@ -210,51 +306,87 @@ class HarnessDriver(Adjust):
...
 
         assert c and c.get(config_name, None), 'Harness Servo Configuration named {} was not provided in "{}". ' \
-                                        'Please refer to README.md'.format(config_name, config_path)
+            'Please refer to README.md'.format(config_name, config_path)
 
         h = c[config_name]
-        assert h.get('opsani_account'), 'No opsani_account was included in harness configuration located at {}'.format(config_path)
-        assert h.get('opsani_app_name'), 'No opsani_app_name was included in harness configuration located at {}'.format(config_path)
-        assert h.get('target_platform'), 'No target_platform was included in harness configuration located at {}'.format(config_path)
+        # What am I going to do with all of these lemons? (Why are these hard crash asserts? Should be a pre-procesing check to verify input data separate, no?)
```
Comment (linkous8) [inline adjust:312]: what did you have in mind? load_config is meant to be the driver's pre-processing check but this was back before we adopted pydantic and a pydantic model could likely streamline this a great deal

File: adjust [hunk 2]
```diff
@@ -15,36 +15,49 @@ VERSION = "0.0.1"
...
+            'min': 0.25,  # vCPU
+            'max': 4,  # vCPU
+            'step': 0.25  # vCPU
+        }
+    },
+    'jvm': {
+        'jvm_gc_time_ratio': {
+            'min': 9,
+            'max': 99,
+            'step': 10,  # Not sure why the docs list this as a step of 10, but it does take us from 9 to 99 without over- or under-shooting
```
Comment (linkous8) [inline adjust:46]: @ekalosak can provide more details but the gist is that it is that too fine a granularity expands the optimization space such that finding a result takes exponentially longer. If we expand the space too much, our normal optimization duration likely wont find an optimal value in time

Comment (benburdick) [reply adjust:46]: Should this be setup in the adjust driver itself like this or should it be a part of user config?

File: adjust [hunk 3]
```diff
@@ -183,7 +237,48 @@ mutation {{
...
+        }}
+      }},
+      {{
+        name: "jvm_parallel_pre_proc_enabled"
+        variableValue: {{
+          type: NAME
+          value: "{jvm_parallel_pre_proc_enabled}"
+        }}
+      }},
     ]
```
Comment (benburdick) [inline adjust:282]: Same with these. Shouldn't we be pulling the values for these out of config (all uppercase, under the correct key, etc)

File: test/test_adjust.py [hunk 4]
```diff
@@ -12,49 +12,63 @@
...
 def test_adjust(monkeypatch):
     with monkeypatch.context() as m:
         # replicate command line arguments fed in by servo
         m.setattr(sys, 'argv', ['', '1234'])
         m.setattr(sys, 'stdin', io.StringIO(adjust_json_stdin))
-        driver = HarnessDriver(cli_desc=DESC, supports_cancel=HAS_CANCEL, version=VERSION)
+        driver = HarnessDriver(
+            cli_desc=DESC, supports_cancel=HAS_CANCEL, version=VERSION)
         driver.run()
         assert True
```
Comment (benburdick) [inline test/test_adjust.py:74]: Please see https://console.opsani.com/accounts/fred.opsani.com/applications/eng-701-3/config?scenario=latest for an example of what the config should look like for a JVM app

File: adjust [hunk 5]
```diff
@@ -317,185 +464,289 @@ class HarnessDriver(Adjust):
...
+        if opsani_config_settings and opsani_config_settings.get('jvm_parallel_ref_proc_enabled'):
+            settings['jvm_parallel_ref_proc_enabled']['value'] = opsani_config_settings['jvm_parallel_ref_proc_enabled']
+        else:
+            settings['jvm_parallel_ref_proc_enabled']['value'] = self.config.get(
+                'jvm_parallel_ref_proc_enabled', {}).get('default')
+
+        return {'application': {'components': {comp_key: {'settings': settings}}}}
 
     def adjust(self, data):
         if self.config.get('adjust_on') and data["control"]["userdata"]["deploy_to"] != self.config['adjust_on']:
```
Comment (linkous8) [inline adjust:538]: See userdata here

**Comment by benburdick** [comment]
This is no longer needed.  Work has been done elsewhere

**Pull request closed without merging**